libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -78,14 +78,11 @@ void UARTDriver::_begin(uint32_t b, uint16_t rxS, uint16_t txS)
 
     while (_in_timer) hal.scheduler->delay(1);
 
-    _device->set_speed(b);
-
     bool clear_buffers = false;
-    if (b != 0) {
-        if (_baudrate != b && hal.console != this) {
-            clear_buffers = true;
-        }
+    if (b != 0 && _baudrate != b && hal.console != this) {
+        _device->set_speed(b);
         _baudrate = b;
```

Comment:
It seems that after this patch,  _baudrate is only set when `hal.console != this`, while it was set regardless in the previous code. could that be an issue?

From a quick glimpse at the code this seems like a harmless change. Thoughts @peterbarker ? 

Other than that this looks good to me.